Detectors/TPC/calibration/include/TPCCalibration/CalibdEdx.h

```diff
@@ -0,0 +1,108 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_TPC_CALIBDEDX_H_
+#define ALICEO2_TPC_CALIBDEDX_H_
+
+#include <string>
+#include <array>
+
+// o2 includes
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DataFormatsTPC/dEdxInfo.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "TPCCalibration/FastHisto.h"
+
+namespace o2
+{
+namespace tpc
+{
+
+// class TrackTPC;
+
+// FIXME: not sure about this
+struct CalibMIPposition {
+  uint64_t timeFrame;
+  std::vector<double> mip;
+};
+
+class CalibdEdx
+{
+  using Hist = FastHisto<float>;
+
+ public:
+  CalibdEdx(double min_p, double max_p, int min_clusters, unsigned int n_bins)
+    : mMinClusters{min_clusters}, mMinP{min_p}, mMaxP{max_p}, mHist{{{n_bins, 0, static_cast<float>(n_bins)}, {n_bins, 0, static_cast<float>(n_bins)}}} // oh my god this is so ugly
+  {
+  }
+
+  void fill(const gsl::span<const TrackTPC> tracks);
+  void merge(const CalibdEdx* prev);
+  void print() const;
+
+  double getASideEntries() const { return mHistEntries[0]; }
+  double getCSideEntries() const { return mHistEntries[1]; }
+
+  void dumpToFile(const std::string& file_name) const;
+  const std::array<Hist, 2>& getHist() const { return mHist; }
+
+ private:
+  int mMinClusters;
+  double mMinP, mMaxP;
```

Comment:
Default initialize, separate lines and comments:
```suggestion
  int mMinClusters{}; ///< bla
  double mMinP{}; ///< bla
  double mMaxP{}; ///< bla
```